cardano-tx-generator/src/Cardano/Benchmarking/GeneratorTx.hs

```diff
@@ -1022,16 +1046,17 @@ txGenerator benchTracer
             txsListsForTargetNodes = do
   liftIO . traceWith benchTracer . TraceBenchTxSubDebug
     $ " Prepare to generate, total number of transactions " ++ show numOfTransactions
+      ++ ", for " ++ show numOfTargetNodes ++ " peers"
 
   -- Generator is producing transactions and writes them in the list.
   -- Later sumbitter is reading and submitting these transactions.
-  txsForSubmission :: MSTM.TMVar IO [CC.UTxO.ATxAux ByteString] <- liftIO $ STM.newTMVarIO []
+  txsForSubmission :: [MSTM.TMVar IO [CC.UTxO.ATxAux ByteString]] <-
+    liftIO $ replicateM numOfTargetNodes $ STM.newTMVarIO []
```

Comment:
so this is not only introducing "cooldown"?
doese this still match the comment?

Reply:
This is a part of the fix for the single-threaded submission bug -- I'll update the comment.